### PR TITLE
Implement the exercise "isbn-verifier"

### PR DESCRIPTION
**Pull request by KevDi**
Implementation of the Exercise ISBN-Verifier.
This is my first Exercise which i ported to the CPP Track.
If wanted i will port more exercises to the CPP Track

### CODE REVIEW

File: config.json
Comment (siebenschlaefer) [inline config.json:734]: Why "lists"?

File: exercises/practice/isbn-verifier/example.cpp
Comment (siebenschlaefer) [inline exercises/practice/isbn-verifier/example.cpp:9]: `[0-9|xX]` includes the character `'|'`.
Also, I'm curious: Why the lowercase `'x'`?

File: exercises/practice/isbn-verifier/example.cpp
Comment (siebenschlaefer) [inline exercises/practice/isbn-verifier/example.cpp:3]: `<vector>` is not used in this solution.

File: exercises/practice/isbn-verifier/isbn_verifier_test.cpp
Comment (siebenschlaefer) [inline exercises/practice/isbn-verifier/isbn_verifier_test.cpp:65]: This is the only name with spaces. Everywhere else they are replaced with dashes.

File: exercises/practice/isbn-verifier/isbn_verifier_test.cpp
Comment (siebenschlaefer) [inline exercises/practice/isbn-verifier/isbn_verifier_test.cpp:2]: In the `_test.cpp` files of the other exercises that `#include` looks like this:
```cpp
#ifdef EXERCISM_TEST_SUITE
#include <catch2/catch.hpp>
#else
#include "test/catch.hpp"
#endif
```

File: exercises/practice/isbn-verifier/example.h
Comment (siebenschlaefer) [inline exercises/practice/isbn-verifier/example.h:2]: I think in the other exercises we use the name of the exercise for the include guard, so `ISBN_VERIFIER_H` instead of `EXAMPLE_H`

**Comment by siebenschlaefer** [comment]
Most of it looks good to me.
I checked the tests cases (`isbn_verifier_test.cpp` and `tests.toml`), they are an exact translation of the `canonical-data.json`. The solution seems to be correct (except for one small detail in a regular expression).
I did not check whether the `config.json` is correct.

**Comment by siebenschlaefer** [comment]
Looks good to me. Now let's wait for the maintainers.

> Is there a better way?

I don't know. There's some tooling that performs validation checks but I never got that to work on my machine.
Usually I just hope that @patricksjackson and the other maintainers will perform those integration and system-level tests.

**Comment by KevDi** [comment]
I moved here also the Example Files to the `.meta` Folder

**Review by kytrinyx — APPROVED** [review]
Thank you so much for implementing the exercise @KevDi.

I'm sorry it's taken so long to get this approved. I'm going to spend some time this week going through all the open PRs in this repository, to see if we can get things into shape.

I made a few tiny tweaks for consistency with other exercises in the track, and did a new `configlet sync` to bring in changes that were made upstream since you opened the PR.

Once CI is happy I'll go ahead and merge this.

**Pull request merged**
